docs/api/index.rst

```diff
@@ -3,16 +3,70 @@
 Python API
 ==========
 
-.. toctree::
-    :maxdepth: 2
+The parsing of a notebook consists of a number of stages, with each stage separated into a separate module:
 
-    nodes
-    render_outputs
+1. The configuration is set (from a file or CLI)
```

Comment:
This is a very helpful description of the whole process, I wonder if we could put this in a place where it is more discoverable? Can do that in a follow up though